src/components/ImageUpload/index.jsx

```diff
@@ -10,22 +10,43 @@ class ImageUpload extends React.Component {
     super(props);
     this.state = {
       value: props.input.value,
+      sizeValidationError: false,
     };
 
     this.handleFilePicked = this.handleFilePicked.bind(this);
+    this.sizeValidator = this.sizeValidator.bind(this);
   }
 
-  updateValue(value) {
-    this.setState({ value });
+  updateValue(value, sizeValidationError = false) {
+    this.setState({ value, sizeValidationError });
     this.props.input.onChange(value);
   }
 
   handleFilePicked(event) {
     const reader = new FileReader();
+    const file = event.target.files[0];
+    // File size is given in bytes
+    const oneMegabyte = 1000000;
+
+    if (file && file.size > oneMegabyte) {
+      this.updateValue('', true);
+      return;
+    }
+
     reader.onload = () => {
       this.updateValue(reader.result);
     };
-    reader.readAsDataURL(event.target.files[0]);
+    reader.readAsDataURL(file);
+  }
+
+  sizeValidator(event) {
+    const { pristine } = this.props.meta;
+
+    if (!pristine && (event.target.naturalWidth !== 1134 || event.target.naturalHeight !== 675)) {
+      this.updateValue('', true);
```

Comment:
I wish JS had Python-style kwarg support. So you could do `sizeValidationError=true` because unlabeled bool variables are really best when they are the only one like `setPristine(true)` or something. With `updateValue('', true)`, the `true` could be for anything.

One way to avoid this would also fix another UX issue - which is that we could do the user the kindness of telling them which error is the problem. So maybe instead of changing `updateValue`, you add a new method like `setValidationError` or something that takes a string and sets value to '' while also setting a state error string (rather than a state boolean). And then when that string is set, you show the StatusAlert.

Eh?